wasmplugin/plugin.go

```diff
@@ -119,18 +143,29 @@ func (ctx *corazaPlugin) OnPluginStart(pluginConfigurationSize int) types.OnPlug
 			return types.OnPluginStartStatusFailed
 		}
 
-		err = perAuthorityWAFs.put(name, waf)
-		if err != nil {
-			proxywasm.LogCriticalf("Failed to register authority WAF: %v", err)
-			return types.OnPluginStartStatusFailed
+		if isTheDefaultWAF {
```

Comment:
I mean, I reread once again and am pretty confident that there is no case this is true while authorities is non-empty, if there is a case like that then the code is not very intuitive I think.

But we have this check and authorities loop inline, so codewise it looks both can be run in the same case. I'm still struggling to see why this isn't just `len(authorities) == 0` even with the extra comments.

Reply:
I just changed it. PTAL @anuraaga